src/main/scala/machines/Internal.scala

```diff
@@ -3,4 +3,31 @@ package machines
 import regex._
 import dfa._
 
-// TODO: Add your code below
+given Conversion[Char, RegularLanguage] =
+    c => Character(c)
+
+given Conversion[String, RegularLanguage] = 
+    s => s.map(c => Character(c)).reduce(Concat(_,_))
```

Comment:
I used recursion for this implementation - I think I understand better what my version is doing, but yours is much cleaner and I can still determine what is happening and how it works.